allensdk/brain_observatory/behavior/behavior_project_cache/project_apis/data_io/behavior_ecephys_project_cloud_api.py

```diff
@@ -0,0 +1,336 @@
+        BehaviorProjectCloudApi instance
+
+        """
+        if use_static_cache:
+            cache = StaticLocalCache(
+                cache_dir,
+                project_name,
+                ui_class_name=ui_class_name
+            )
+        else:
+            cache = LocalCache(
+                cache_dir,
+                project_name,
+                ui_class_name=ui_class_name
+            )
+        return BehaviorEcephysProjectCloudApi(cache, local=True)
+
+    def get_behavior_session(
+            self, behavior_session_id: int) -> BehaviorSession:
+        """get a BehaviorSession by specifying behavior_session_id
+
+        Parameters
+        ----------
+        behavior_session_id: int
+            the id of the behavior_session
+
+        Returns
+        -------
+        BehaviorSession
+
+        Notes
+        -----
+        behavior session does not include file_id.
+        The file id is accessed via ecephys_session_id key
+        from the ecephys_session_table
+        """
+        row = self._behavior_session_table.query(
+                f"behavior_session_id=={behavior_session_id}")
+        if row.shape[0] != 1:
+            raise RuntimeError("The behavior_session_table should have "
+                               "1 and only 1 entry for a given "
+                               "behavior_session_id. For "
+                               f"{behavior_session_id} "
+                               f" there are {row.shape[0]} entries.")
+        row = row.squeeze()
+        esid = row.ecephys_session_id
+        row = self._ecephys_session_table.query(f"index=={esid}")
+        file_id = str(int(row[self.cache.file_id_column]))
+        data_path = self._get_data_path(file_id=file_id)
+        return BehaviorSession.from_nwb_path(str(data_path))
+
+    def get_ecephys_session(
+        self,
+        ecephys_session_id: int
+    ) -> BehaviorEcephysSession:
+
+        """get a BehaviorEcephysSession by specifying ecephys_session_id
+
+        Parameters
+        ----------
+        ecephys_session_id: int
+            the id of the ecephys session
+
+        Returns
+        -------
+        BehaviorEcephysSession
+
+        """
+        row = self._ecephys_session_table.query(
+                f"index=={ecephys_session_id}")
+        if row.shape[0] != 1:
+            raise RuntimeError("The behavior_ecephys_session_table should "
+                               "have 1 and only 1 entry for a given "
+                               f"ecephys_session_id. For "
+                               f"{ecephys_session_id} "
+                               f" there are {row.shape[0]} entries.")
+        file_id = str(int(row[self.cache.file_id_column]))
+        data_path = self._get_data_path(file_id=file_id)
+        return BehaviorEcephysSession.from_nwb_path(
+            str(data_path))
+
+    def _get_ecephys_session_table(self):
+        session_table_path = self._get_metadata_path(
+            fname="sessions")
```

Comment:
let's ask Corbett if we shouldn't just rename `sessions.csv` -> `ecephys_sessions.csv`. It seems more descriptive.